moveit_ros/move_group/src/default_capabilities/execute_path_action_capability.cpp

```diff
@@ -0,0 +1,140 @@
+#include <moveit/move_group/capability_names.h>
+
+namespace move_group {
+
+MoveGroupExecutePathAction::MoveGroupExecutePathAction() :
+  MoveGroupCapability("ExecutePathAction")
+{
+}
+
+void MoveGroupExecutePathAction::initialize()
+{
+  // start the move action server
+  execute_action_server_.reset(new actionlib::SimpleActionServer<moveit_msgs::ExecutePathAction>
+                               (root_node_handle_, EXECUTE_ACTION,
+                                boost::bind(&MoveGroupExecutePathAction::executePathCallback, this, _1), false));
+  execute_action_server_->registerPreemptCallback(boost::bind(&MoveGroupExecutePathAction::preemptExecutePathCallback, this));
+  execute_action_server_->start();
+}
+
+void MoveGroupExecutePathAction::executePathCallback(const moveit_msgs::ExecutePathGoalConstPtr& goal)
+{
+  moveit_msgs::ExecutePathResult action_res;
+  if (!context_->trajectory_execution_manager_)
+  {
+    std::string response = "Cannot execute trajectory since ~allow_trajectory_execution was set to false";
+    action_res.error_code.val = moveit_msgs::MoveItErrorCodes::CONTROL_FAILED;
+    execute_action_server_->setAborted(action_res, response);
+    return;
+  }
+
+  executePathCallback_Execute(goal, action_res);
+
+  std::string response = getActionResultString(action_res.error_code, false, false);
+  if (action_res.error_code.val == moveit_msgs::MoveItErrorCodes::SUCCESS)
+  {
+    execute_action_server_->setSucceeded(action_res, response);
+  }
+  else if (action_res.error_code.val == moveit_msgs::MoveItErrorCodes::PREEMPTED)
+  {
+    execute_action_server_->setPreempted(action_res, response);
+  }
+  else
+  {
+    execute_action_server_->setAborted(action_res, response);
+  }
+
+  setExecutePathState(IDLE);
+}
+
+void MoveGroupExecutePathAction::executePathCallback_Execute(const moveit_msgs::ExecutePathGoalConstPtr& goal, moveit_msgs::ExecutePathResult &action_res)
+{
+  ROS_INFO("Execution request received for ExecutePath action.");
+
+  context_->trajectory_execution_manager_->clear();
+  if (context_->trajectory_execution_manager_->push(goal->trajectory))
+  {
+    setExecutePathState(MONITOR);
+    context_->trajectory_execution_manager_->execute();
+    moveit_controller_manager::ExecutionStatus es = context_->trajectory_execution_manager_->waitForExecution();
+    if (es == moveit_controller_manager::ExecutionStatus::SUCCEEDED)
+    {
+      action_res.error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+    }
+    else if (es == moveit_controller_manager::ExecutionStatus::PREEMPTED)
+    {
+      action_res.error_code.val = moveit_msgs::MoveItErrorCodes::PREEMPTED;
+    }
+    else if (es == moveit_controller_manager::ExecutionStatus::TIMED_OUT)
+    {
+      action_res.error_code.val = moveit_msgs::MoveItErrorCodes::TIMED_OUT;
+    }
+    else
+    {
+      action_res.error_code.val = moveit_msgs::MoveItErrorCodes::CONTROL_FAILED;
+    }
+    ROS_INFO_STREAM("Execution completed: " << es.asString());
```

Comment:
would prefer ``_NAMED`` for the console log, see bottom of http://moveit.ros.org/documentation/contributing/code.html